Source/WebCore/page/LocalFrameView.cpp

```diff
@@ -3163,10 +3164,13 @@ void LocalFrameView::scrollRectToVisibleInTopLevelView(const LayoutRect& absolut
 
     // Avoid scrolling to the rounded value of revealRect.location() if we don't actually need to scroll
     if (revealRect != viewRect) {
-        // FIXME: Should we use document()->scrollingElement()?
-        // See https://bugs.webkit.org/show_bug.cgi?id=205059
-        ScrollOffset clampedScrollPosition = roundedIntPoint(revealRect.location()).constrainedBetween(minScrollPosition, maxScrollPosition);
-        setScrollPosition(clampedScrollPosition, scrollPositionChangeOptionsForElement(*this, element, options));
+        if (options.onlyAllowForwardScrolling == OnlyAllowForwardScrolling::No || (revealRect.y() > viewRect.y())) {
```

Comment:
`OnlyAllowForwardScrolling::No` is a little hard to parse and sounds like it should prevent something; this would be more readable as `BackwardsScrollingAllowed::Yes || || (revealRect.y() > viewRect.y())`

Reply:
Went back and forth on this, but we really need a negative to make us not have to add this flag everywhere. Leaving the name for now.